record.py

```diff
@@ -87,28 +89,38 @@ def init_plot(self):
         self.axes = self.fig.add_subplot(111)
 
 
-    def on_timer(self, event):
+    def on_timer(self):
         self.poll()
 
         # stop drawing if recording to avoid slow downs
         if self.recording == False:
             self.draw()
 
+        if not self.pause_timer:
+            self.root.after(self.rate, self.on_timer)
+
 
     def poll(self):
-        self.bmp = self.take_screenshot()
+        self.img = self.take_screenshot()
         self.controller_data = self.controller.read()
         self.update_plot()
 
         if self.recording == True:
             self.save_data()
+            self.t += 1
+
 
     def take_screenshot(self):
-        screen = wx.ScreenDC()
-        bmp = wx.Bitmap(Screenshot.SRC_W, Screenshot.SRC_H)
-        mem = wx.MemoryDC(bmp)
-        mem.Blit(0, 0, Screenshot.SRC_W, Screenshot.SRC_H, screen, Screenshot.OFFSET_X, Screenshot.OFFSET_Y)
-        return bmp
+        with mss.mss() as sct:
```

Comment:
For better performances, you could assign it to a class attribute. With MSS version 3.2.0 (just released), I benchmarked and it apperas that avoiding instanciation is a **huge** improvement.

Reply:
Ah yes, thanks for pointing that out. I think I read that yesterday, but forgot to move it. In my other project I'm just instantiating once and holding onto the reference. I'll do the same here in a bit.

Reply:
Wow, you weren't kidding about it ***just*** being released. Quite literally just before you posted your comment. And you've already added this project to the list. You really don't waste any time :)

Reply:
I've now upgraded to `3.2.0`, utilized the `.bgra` property, and moved instantiation to the `__init__` method so we only do it once. Thanks for the comment :+1:

Reply:
Nice! :)

Reply:
This thread makes me so happy! Thanks for keeping us in the loop about updates and changes to mss :heart::heart::heart: